sotodlib/io/load_smurf.py

```diff
@@ -265,6 +265,18 @@ def add_file(self, path, session, overwrite=False):
             db_frames = db_file.frames
             [session.delete(frame) for frame in db_frames]
             session.commit()
+        
+        status = SmurfStatus.from_file(db_file.name)
+
+        if status.tune is None or status.tune == "":
+            my_tune = None
+        else:
+            my_tune = session.query(Tunes).filter(
+                Tunes.name==Tunes.get_name_from_status(status)
+            ).one_or_none()
+        if my_tune is not None:
+            ## tune has been found in streamed data. 
+            self.add_tuneset_to_file(my_tune, db_file, session)
 
```

Comment:
Does this create the requirement that the Tune must exist before the g3 file is indexed? I feel like this is kind of problematic, especially since it's very possible for a G3 file to be synced before the corresponding tune is synced (since there's a separate suprsync agent for the g3 timestreams and the smurf files). This means that even if `index_metadata`and `index_archive` are called in the correct order there can definitely still be cases where a g3-file is indexed before its corresponding tune...

In the case where `my_tune is None` can we create an empty tune that is populated when the metadata is indexed?

Reply:
It does create that requirement but that's why I changed the order of indexing in the update script. The `index_metadata` runs through just channel assignment and tune files and adds all of them to the database. So if all the data transfer is up-to-date then this will always work. 

If the data transfer stops at some point we'll have the monitor telling us observations don't have readout ids. There is a way to re-run the updates to fix those. Although I've just realized I should add one more check to the update script to catch them automatically.

Reply:
ok, new commit will let us re-check for tunes if data transfer between `smurf` and `timestreams` gets delayed.

Reply:
Nice, I still don't love the soft requirement that the data transfer is up-to-date since that adds some strange race conditions, but if exceptions are automatically handled by the update script that's good enough for me!

Reply:
yeah but there have always been these race conditions they were just in another place. Misses can be dealt with automatically and this way we have a lot lower overhead on the number of tunesets we need to track / make readout/detector id matching for.